package.json

```diff
@@ -3,25 +3,35 @@
   "version": "3.0.2",
   "description": "Utilities for dealing with cross-domain windows.",
   "main": "dist/cross-domain-utils.js",
-  "module": "dist/module/index.js",
+  "module": "lib/index.js",
+  "types": "dist/types/index.d.ts",
+  "exports": {
+    ".": {
+      "import": "./dist/my-component-lib.es.js",
+      "require": "./dist/my-component-lib.umd.js"
+    }
+  },
   "scripts": {
-    "build": "npm run test && npm run babel && npm run webpack",
+    "build": "npm run test && npm run babel && npm run webpack && npm run build:types",
+    "build:flow": "find ./dist -type f -not -path './node_modules/*' -name '*.d.ts' -exec sh -c 'flowgen --add-flow-header $1 -o ${1%.*.*}.js.flow' _ '{}' \\;",
+    "build:tsc": "tsc src/* --outDir ./dist/esm --declaration --emitDeclarationOnly",
+    "build:types": "npm run build:tsc && npm run build:flow",
     "webpack": "cross-env NODE_ENV=production babel-node --plugins=transform-es2015-modules-commonjs ./node_modules/.bin/webpack --progress --output-path dist",
-    "babel": "cross-env NODE_ENV=production babel src/ --out-dir dist/module",
-    "karma": "cross-env NODE_ENV=test babel-node --plugins=transform-es2015-modules-commonjs ./node_modules/.bin/karma start",
+    "babel": "cross-env NODE_ENV=production babel src/ --out-dir ./dist/esm/ --extensions .ts,.tsx",
+    "tsc": "tsc",
     "format": "prettier --write --ignore-unknown .",
     "format:check": "prettier --check .",
-    "test": "npm run format:check && npm run lint && npm run flow-typed && npm run flow && npm run karma",
-    "lint": "eslint src/ test/",
-    "flow": "flow",
-    "flow-typed": "rm -rf ./flow-typed && flow-typed install",
+    "test": "npm run format:check && npm run lint && npm run tsc --no-emit && npm run vitest",
```

Comment:
```
  npm run tsc --no-emit
```
does that work without the extra `--`? like `npm run tsc -- --no-emit`

Reply:
yeah works just fine without